### ISSUE TITLE
FK Relationship View constraint name natural ordering

### ISSUE THREAD

**Issue by cjhewett** [phpmyadmin/phpmyadmin#17182]
### Is your feature request related to a problem? Please describe.

FKs are assigned an automatic name which ends with a numeric suffix `_1`, `_2` etc, Once this goes over `_10` PMAs orders them e.g. 1, 10, 2, 3 etc. 

### To Replicate

```sql
# create a table
CREATE TABLE `2_test` (
  `id` tinyint(3) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB;

# create another table with 10+ foreign keys
CREATE TABLE `1_test` (
  `id` tinyint(3) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
  CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`)
) ENGINE=InnoDB;
```
View `1_test` Structure > Relationship View

![image](https://user-images.githubusercontent.com/13466620/140909801-3320cf74-522d-4be3-a7b2-9ec423323244.png)

### Describe the solution you'd like

The ordering should be in the more natural 1, 2, 10, 11, 20 so the FKs are listed in the order they were created by default.

### Additional context

- phpMyAdmin version: 5.1.1-1

**Comment by ibennetch** [comment]
I agree that this is annoying. However, it's the order MySQL (or at least MariaDB 10.5 in my case) is returning the results. 

It may be possible for us to cast the column in order to force a more natural sorting, but I'm no sure offhand whether that's going to be desirable (because then our behavior will differ from MySQL's). I'm open to the idea.

**Comment by ChrisHSandN** [comment]
There is a [NATURAL_SORT_KEY](https://mariadb.com/kb/en/natural_sort_key/) coming in MariaDB 10.7, maybe I asked a few years too early! I was imagining something based around PHP `natsort`.

Personally I don't expect the PMA "Relationship View" to be an equivalent view of the `KEY_COLUMN_USAGE` table etc. so I wouldn't expect the same ordering. However I can appreciate that [might not be the case for everyone](https://xkcd.com/1172/).

**Comment by hameddanesh** [comment]
> 
> ### Is your feature request related to a problem? Please describe.
> 
> FKs are assigned an automatic name which ends with a numeric suffix `_1`, `_2` etc, Once this goes over `_10` PMAs orders them e.g. 1, 10, 2, 3 etc.
> ### To Replicate
> 
> ```
> # create a table
> CREATE TABLE `2_test` (
>   `id` tinyint(3) unsigned NOT NULL AUTO_INCREMENT,
>   PRIMARY KEY (`id`)
> ) ENGINE=InnoDB;
> 
> # create another table with 10+ foreign keys
> CREATE TABLE `1_test` (
>   `id` tinyint(3) unsigned NOT NULL AUTO_INCREMENT,
>   PRIMARY KEY (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`),
>   CONSTRAINT FOREIGN KEY (`id`) REFERENCES `2_test` (`id`)
> ) ENGINE=InnoDB;
> ```
> 
> View `1_test` Structure > Relationship View
> 
> ![image](https://user-images.githubusercontent.com/13466620/140909801-3320cf74-522d-4be3-a7b2-9ec423323244.png)
> ### Describe the solution you'd like
> 
> The ordering should be in the more natural 1, 2, 10, 11, 20 so the FKs are listed in the order they were created by default.
> ### Additional context
> 
>     * phpMyAdmin version: 5.1.1-1

this happens because mysql is ordering keys in a digit by digit manner. for example 100 will always be above 20 because  '2'>'1' (in asscii or .etc) and from machine logic it makes sense because these names are treated as strings
you can add some zeros before numbers to fix this for now. like 001, 002 ..., 010 and please report it on mysql/mysql-server to be maintained

**Comment by ibennetch** [comment]
I'd like to see us fix this, although it's certainly lower priority. https://stackoverflow.com/questions/8557172/mysql-order-by-sorting-alphanumeric-correctly has some ideas about how to improve the sorting, but it seems using PHP's `natsort()` is going to be best IMHO.

This might make a good first issue for someone looking to apply for GSoC with us.

**Comment by aonamrata** [comment]
@ibennetch  https://github.com/phpmyadmin/phpmyadmin/pull/19323 is my attempt yo fix it at the php end instead of changing the api call.  Please review 🙏
